core/trino-main/src/main/java/io/trino/spiller/NodeSpillConfig.java

```diff
@@ -14,16 +14,21 @@
 package io.trino.spiller;
 
 import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
 import io.airlift.configuration.LegacyConfig;
 import io.airlift.units.DataSize;
+import io.trino.execution.buffer.CompressionCodec;
 import jakarta.validation.constraints.NotNull;
 
+import static io.trino.execution.buffer.CompressionCodec.LZ4;
+
 public class NodeSpillConfig
 {
     private DataSize maxSpillPerNode = DataSize.of(100, DataSize.Unit.GIGABYTE);
     private DataSize queryMaxSpillPerNode = DataSize.of(100, DataSize.Unit.GIGABYTE);
 
     private boolean spillCompressionEnabled;
```

Comment:
We should deprecate the boolean flag and use the enum, NONE should be the new replacement for boolean flag. This would consistent with how `io.trino.plugin.hive.HiveCompressionOption` works